kernel/priority_queues.c

```diff
@@ -94,16 +94,18 @@ struct k_thread *z_priq_rb_best(struct _priq_rb *pq)
 
 struct k_thread *z_priq_mq_best(struct _priq_mq *pq)
 {
-	if (!pq->bitmask) {
-		return NULL;
-	}
+	for (int i = 0; i < K_NUM_THRAD_PRIO; ++i) {
+		if (!pq->bitmask[i]) {
```

Comment:
We are iterating through the loop the wrong number of times--this has the potential to exceed the bounds of bitmask[]. Better to stop at PRIQ_BITMAP_SIZE.

Reply:
Yeah, this is wrong.  To state it differently: the size of the bitmask[] array is 1/32 the number of priorities, not the number of priorities.  This is going to run right off the end of the array.